### PR TITLE
[TS-37] Append Events & Stop duplicate calls for events

### PR DESCRIPTION
**Pull request by William-Unosquare**
Events now get appended when new dashboard event requests are made. A request will be made if a month doesn't have any events, and a request won't be made if events for that month already exist. There's also checks to avoid mandatory holidays blocking requests.

See comments below for more info.

### CODE REVIEW

File: web-app/src/actions/dashboard.js [hunk 1]
```diff
@@ -63,30 +64,37 @@ export const setEventView = eventView => {
...
+        })
+        .catch(error => {
+          dispatch(setLoading(false));
+          dispatch(setError(error));
+        });
+      // Fetch team's events as well as your own.
+    } else if (eventView === eventsView.TEAM_EVENTS) {
+      getTeamsEvents(date)
+        .then(({ data }) => {
+          dispatch(setLoading(false));
```
Comment (ghost) [inline web-app/src/actions/dashboard.js]: The fact the code in the block on line 89 onwards plus line 76 onwards is the same....maybe we could pull this out and reuse it within a function?

File: web-app/src/utilities/dashboardEvents.js [hunk 2]
```diff
@@ -49,3 +70,29 @@ export const getEventDuration = event => {
...
+  }, []);
+
+  // Check if any of the dates' month match the month the user is viewing. If
+  // there's a match, we don't want to update.
+  eventDates.forEach(date => {
+    if (date.month() === month) {
+      requireNewRequest = false;
+    }
+  });
+  console.log('reqreq', requireNewRequest);
```
Comment (ghost) [inline web-app/src/utilities/dashboardEvents.js]: A log has been left in

**Review by ghost — CHANGES_REQUESTED** [review]
Just a question plus comment

File: web-app/src/actions/dashboard.js (hunk 1 above)
Comment (William-Unosquare) [reply web-app/src/actions/dashboard.js]: Make sense. Will do. 👍

File: web-app/src/utilities/dashboardEvents.js (hunk 2 above)
Comment (William-Unosquare) [reply web-app/src/utilities/dashboardEvents.js]: Well spotted.

**Comment by William-Unosquare** [comment]
This is working well now I think. This may be worth pulling down and actually giving a test yourselves to make sure I haven't missed any edge cases.

The expected behaviour is:
- Events should be pushed into a state when navigating months.
- When a month is empty, it fires off a request to retrieve events.
- When there are already events for the month being viewed in state, it will not fire a request for events.
- When a user toggles team view or creates an event we force an update. This means all events are removed from state, and we request updated event data for the viewed month.

**Review by ghost — APPROVED** [review]

File: web-app/src/utilities/AxiosInstance.js [hunk 3]
```diff
@@ -55,6 +55,19 @@ instance.interceptors.response.use(function(response) {
     };
   }
 
+  if (response.config.url.includes(`${baseURL}/dashboard/getTeamEvents`)) {
```
Comment (paulmatchett) [inline web-app/src/utilities/AxiosInstance.js:58]: Is this axios middleware required? Does the _formatEventsForCalendar method in dashBoardEvents.js not convert the dates to moments too?

**Review by paulmatchett — COMMENTED** [review]
Flows great, comments used well and is easy to understand. Only one query and that was regarding the axios middleware for teams call

File: web-app/src/utilities/AxiosInstance.js (hunk 3 above)
Comment (William-Unosquare) [reply web-app/src/utilities/AxiosInstance.js:58]: Good spot. I think the issue here is actually in the dashboardEvents. I'd rather keep the place we transform dates into moments in the AxiosInstance in case the call was ever used elsewhere in the app. Just keeps it consistent too. 

Pushed a fix.

**Review by paulmatchett — APPROVED** [review]
